starport/services/networkbuilder/networkbuilder.go

```diff
@@ -456,6 +458,48 @@ func (b *Builder) StartChain(ctx context.Context, chainID string, flags []string
 	return g.Wait()
 }
 
+// Generate the genesis from the launch information in a temporary directory and return this directory
+func (b *Builder) GenerateTemporaryGenesis(ctx context.Context, chainID string, homeDir string, launchInfo spn.LaunchInformation) (string, error) {
+	tmpHome, err := ioutil.TempDir("", chainID+"*")
+	if err != nil {
+		return "", err
+	}
+
+	chainInfo, err := b.ShowChain(ctx, chainID)
+	if err != nil {
+		return "", err
+	}
+
+	appPath := filepath.Join(sourcePath, chainID)
+	chainHandler, err := chain.New(ctx, appPath,
+		chain.HomePath(tmpHome),
+		chain.LogLevel(chain.LogSilent),
+		chain.KeyringBackend(chaincmd.KeyringBackendTest),
+	)
+	if err != nil {
+		return "", err
+	}
+
+	// copy the default config to the temporary directory
+	originHome := homeDir
+	if originHome == "" {
+		originHome, err = chainHandler.DefaultHome()
+		if err != nil {
+			return "", err
+		}
+	}
+	if err := copy.Copy(originHome, tmpHome); err != nil {
```

Comment:
Couldn't get why this step is needed since `chain.HomePath(tmpHome)` option already passed to `chain.New`?

Reply:
`chain.HomePath(tmpHome)` is passed to `chain.New` to run the genesis generation commands inside the temporary directory but this temporary directory doesn't' have the initial home config at first

Reply:
I think is somehow relavent with https://github.com/tendermint/starport/pull/896#discussion_r599697287?